backend/azure/dtmb/dtmb.go

```diff
@@ -0,0 +1,531 @@
+	}
+
+	return ret, nil
+}
+
+// CompleteOrchestrationWorkItem completes a work item by saving the updated runtime state to durable storage.
+//
+// Returns [ErrWorkItemLockLost] if the work-item couldn't be completed due to a lock-lost conflict (e.g., split-brain).
+func (d *dtmb) CompleteOrchestrationWorkItem(_ context.Context, item *backend.OrchestrationWorkItem) error {
+	completionToken := item.Properties["CompletionToken"].(string)
+	orchestrationName := item.Properties["OrchestrationName"].(string)
+	version := item.Properties["Version"].(string)
+
+	d.connectWorkerClientStream <- &dtmbprotos.ConnectWorkerClientMessage{
+		Message: &dtmbprotos.ConnectWorkerClientMessage_CompleteOrchestration{
+			CompleteOrchestration: &dtmbprotos.CompleteOrchestrationMessage{
+				OrchestrationId: string(item.InstanceID),
+				Name:            orchestrationName,
+				Version:         version,
+				CompletionToken: completionToken,
+				CustomStatus:    item.State.CustomStatus.GetValue(),
+				Actions:         nil, // // what do I put here?
+			},
+		},
+	}
+	// this is a terminal event, so we can evict the cache
+	d.orchestrationHistoryCache.EvictCacheForOrchestrationID(string(item.InstanceID))
+	return nil
+}
+
+// AbandonOrchestrationWorkItem undoes any state changes and returns the work item to the work item queue.
+//
+// This is called if an internal failure happens in the processing of an orchestration work item. It is
+// not called if the orchestration work item is processed successfully (note that an orchestration that
+// completes with a failure is still considered a successfully processed work item).
+func (d *dtmb) AbandonOrchestrationWorkItem(_ context.Context, item *backend.OrchestrationWorkItem) error {
+	completionToken := item.Properties["CompletionToken"].(string)
+	orchestrationName := item.Properties["OrchestrationName"].(string)
+	version := item.Properties["Version"].(string)
+
+	d.connectWorkerClientStream <- &dtmbprotos.ConnectWorkerClientMessage{
+		Message: &dtmbprotos.ConnectWorkerClientMessage_AbandonOrchestration{
+			AbandonOrchestration: &dtmbprotos.AbandonOrchestrationMessage{
+				OrchestrationId: string(item.InstanceID),
+				Name:            orchestrationName,
+				Version:         version,
+				CompletionToken: completionToken,
+			},
+		},
+	}
+	// this is a terminal event, so we can evict the cache
+	d.orchestrationHistoryCache.EvictCacheForOrchestrationID(string(item.InstanceID))
+	return nil
+}
+
+// GetActivityWorkItem gets a pending activity work item from the task hub or returns [ErrNoWorkItems]
+// if there are no pending activity work items.
+func (d *dtmb) GetActivityWorkItem(context.Context) (*backend.ActivityWorkItem, error) {
+	var ret *backend.ActivityWorkItem = nil
+	item := d.activityQueue.Dequeue()
+	if item == nil {
+		return nil, backend.ErrNoWorkItems
+	}
+
+	// this can't be the right way to provide the input, but how should it be done?
+
+	genericInputWrapper := protos.HistoryEvent{
```

Comment:
The code that executes the activity work item is expecting a `TaskScheduledEvent`.

Reply:
Ok great - I had that before but then wasn't sure.